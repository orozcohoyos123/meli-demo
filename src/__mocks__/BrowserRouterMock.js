import React from 'react';
import { BrowserRouter, Switch } from 'react-router-dom';
import Layout from '../components/Layout';

const BrowserRouterMock = props => (
    <BrowserRouter>
        <Layout>
            <Switch>
                {props.children}
            </Switch>
        </Layout>
    </BrowserRouter>
);

export default BrowserRouterMock;
