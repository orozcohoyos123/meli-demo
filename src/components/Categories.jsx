import React from "react";
import "./styles/Categories.scss";
import NextLogo from "../assets/ic_next.png";
import { Link } from "react-router-dom";

const Categories = (props) => {
 
  return (
    <div className="categories col-11">
      <div className="categories__container">
        {props.items && props.items.map((item, index) => {
          const lastItem = props.items.length - 1 === index;
          return (
            <React.Fragment key={`${item.id}_${index}`}>
              <Link to={`/items?search=${item.name}`}>
                <span
                  className={`categories__item ${lastItem ? "bold" : ""}`}
                >
                  {item.name}
                </span>
              </Link>
              {!lastItem && (
                <img 
                  src={NextLogo}
                  className="categories__separator"
                  alt="Next"
                />
              )}
            </React.Fragment>
          );
        })}
      </div>
    </div>
  );
};

export default Categories;
