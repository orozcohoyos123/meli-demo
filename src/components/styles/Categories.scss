@import '../../styles/CommonColors.scss';
@import '../../styles/CommonFontSize.scss';

.bold {
  color: $gray;
  font-weight: bold;
}

.categories {
  padding: 16px 0;
}

.categories__item {
  color: $middle-gray;
  cursor: pointer;
  font-size: $small;
}

.categories__separator {
  height: 14px;
  margin: 0 5px 2px 5px;
}
