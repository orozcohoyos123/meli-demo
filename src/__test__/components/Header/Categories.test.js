import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer'
import Categories from '../../../components/Categories';
import BrowserRouterMock from '../../../__mocks__/BrowserRouterMock';

const mockCategories = [
    {id: "MLA1144", name: "Consolas y Videojuegos"},
    {id: "MLA438566", name: "Consolas"}
]

describe('<Categories />', () => {
    test('Debe renderizar el componente Categories con mock de categorias', () => {
        const categories = mount(
            <BrowserRouterMock>
                <Categories items={mockCategories} />
            </BrowserRouterMock>
        );
        
        expect(categories.length).toEqual(1);
    })

    test('Debe fallar el render del componente Categories sin mock de categorias', () => {
        try {
            const categories = mount(
                <BrowserRouterMock>
                    <Categories />
                </BrowserRouterMock>
            );
    
            expect(true).toBe(false);
        } 
        catch (e) {
            expect(e).toEqual(new Error("Cannot read property 'map' of undefined"));
        }
    })
});

describe('Categories sanpshot', () => {
    test('Comprobar UI componente Categories', () => {
        const categories = 
            create(
                <BrowserRouterMock>
                    <Categories items={mockCategories} />
                </BrowserRouterMock>
            );
        
        expect(categories.toJSON()).toMatchSnapshot();
    })
});
