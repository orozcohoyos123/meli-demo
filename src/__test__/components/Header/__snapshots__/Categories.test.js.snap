// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories sanpshot Comprobar UI componente Categories 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="header__wrapper row"
    >
      <a
        className="header__logo col-2 col-md-1"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Meli"
          src="Logo_ML.png"
        />
      </a>
      <div
        className="searchBox col-9 col-md-10"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            name="query"
            placeholder="Nunca dejes de buscar"
            type="text"
          />
          <button
            type="submit"
          >
            <img
              alt="Search"
              src="ic_Search.png"
            />
          </button>
        </form>
      </div>
    </div>
  </header>,
  <div
    className="categories col-11"
  >
    <div
      className="categories__container"
    >
      <a
        href="/items?search=Consolas y Videojuegos"
        onClick={[Function]}
      >
        <span
          className="categories__item "
        >
          Consolas y Videojuegos
        </span>
      </a>
      <img
        alt="Next"
        className="categories__separator"
        src="ic_next.png"
      />
      <a
        href="/items?search=Consolas"
        onClick={[Function]}
      >
        <span
          className="categories__item bold"
        >
          Consolas
        </span>
      </a>
    </div>
  </div>,
]
`;
